source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -251,11 +251,14 @@ void ShaderGraph::addDefaultGeomNode(ShaderInput* input, const GeomPropDef& geom
 void ShaderGraph::addColorTransformNode(ShaderInput* input, const ColorSpaceTransform& transform, GenContext& context)
 {
     ColorManagementSystemPtr colorManagementSystem = context.getShaderGenerator().getColorManagementSystem();
-    if (!colorManagementSystem)
+    if (!colorManagementSystem || input->getConnection())
```

Comment:
Add a check for input connections here, where all input connections have been created.